source/io/module_javaScriptInvoke.js

```diff
@@ -349,65 +348,26 @@
             REJECTED: 'REJECTED'
         };
 
-        var updateReturnValue = function (
-            functionName,
-            returnTypeName,
-            returnValuePointer,
-            returnUserValue,
-            errorValueRef) {
-            if (returnTypeName === 'StaticTypeAndData') {
-                // User doesn't want return value. If we're passing '*' for the return in VIA code, we get StaticTypeAndData
+        var updateReturnValue = function (functionName, returnValueRef, returnUserValue, errorValueRef) {
+            if (returnValueRef.dataRef === 0) {
```

Comment:
Ah I get it. So this would be true if the user configures the JSLI call to not return. So we say, even if ultimately the invoked function returned something in `returnUserValue` we are just going to ignore it, right?
Please use the variable `NULL` here to indicate that comes from C++ or if you decide to change the behavior of `createValueRef` to return `undefined` then don't forget to update this one.

Reply:
We should talk about this more but I think it is equally valid for returnValueRef to just be undefined. Having access to the type does not actually add value here.

Reply:
Explained in other place. This check is changed.